redfish/eventdestination.go

```diff
@@ -7,6 +7,7 @@ package redfish
 import (
 	"encoding/json"
 	"fmt"
+	oemZt "github.com/stmcginnis/gofish/oem/zt"
```

Comment:
Common code should not import the OEM code.

Reply:
The function [CreateEventSubscription](https://github.com/stmcginnis/gofish/blob/main/redfish/eventservice.go#L302) is the common call to be used on all three vendors.
This function warps the [CreateEventDestination](https://github.com/stmcginnis/gofish/blob/46c07e7c05bda4b260b98647ab3c379c627bdd87/redfish/eventdestination.go#L342) which uses a http request payload that ZT systems does not accept.
ZT systems only accepts this [payload](https://github.com/nwaizer/gofish/blob/ZTsystem/oem/zt/zt.go#L17)

The [oem check](https://github.com/stmcginnis/gofish/blob/46c07e7c05bda4b260b98647ab3c379c627bdd87/redfish/eventdestination.go#L377) only adds the oem data to the existing payload.
I need a way to allow me to change the payload in case it is the vendor ZT, before you issue a post with it.

So I used:
```
	// ZT systems uses a different payload for this action
	if oem == "ZT" {
		return oemZt.SubscribeZT(c, uri, destination, protocol, context)
	}
```

Reply:
will do after answer for https://github.com/stmcginnis/gofish/pull/186#discussion_r913713326